src/pages/components/text-input/usage.mdx

```diff
@@ -132,99 +152,365 @@ reflect the length of the content you expect the user to enter.
+| ----------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| _Enabled_   | When a text input is live but a user is not directly interacting with it. This is commonly referred to as the default or normal state of the component. An enabled text input field can contain no content, placeholder text, or user generate content.           |
+| _Active_    | When a user is actively typing content into the field.                                                                                                                                                                                                            |
+| _Focus_     | When a user `tabs` to or clicks on the text input, it becomes focused, indicating the user has successfully navigated to the component.                                                                                                                           |
+| _Error_     | When the user input is invalid or a required text input has not been filled in. It can also be triggered due to a system error. This state requires a user response before data can be submitted or saved.                                                        |
+| _Warning_   | When you need to call the user's attention to an exception condition. The condition might not be an error but can cause problems if not resolved.                                                                                                                 |
+| _Disabled_  | When the user is not allowed to interact with the text input due to either permissions, dependencies, or pre-requisites. The disabled state completely removes the interactive function from a component. The styling is not subject to WCAG contrast compliance. |
+| _Skeleton_  | Use on an initial page load to indicate that the text input has not yet fully loaded.                                                                                                                                                                             |
+| _Read-only_ | _Coming soon!_                                                                                                                                                                                                                                                    |
+
+<Row>
+<Column colLg={12}>
+
+![Text input states](images/text-input-states.png)
+
+</Column>
+</Row>
+
+### Validation
+
+Real-time validation helps streamline the process and keep data clean when the
+user fills out forms. Otherwise, validate the text input data when the user
+submits the associated form. For complete guidance on error and validation, see
+the form pattern.
```

Comment:
```suggestion
the [form pattern](https://carbondesignsystem.com/patterns/forms-pattern/).
```
Suggestion to link out to the form pattern here to make it easier for people to find.